### ISSUE TITLE
Gym bouldering video record feature

### ISSUE THREAD

**Issue by ellen-goc** [weyucou/hangboard-trainer#7]
## Summary

Add a gym bouldering video record feature: record climbing attempts from within the app so climbers can capture, review, and store their boulder attempts alongside their training data.

## Motivation

The app currently covers hangboard training only. Recording gym attempts extends it toward a full climbing-session companion and is the foundation for two follow-up features: route identification from video and auto-edit/post to Instagram.

## Proposed Scope

- In-app camera recording (`expo-camera`) with start/stop, front/back toggle, tied to a gym attempt
- Attempt metadata: date/time (auto), gym name (free text), route grade (user-entered), send/attempt result
- Video files stored on the device filesystem (`expo-file-system` document directory); **only URIs + metadata persisted to AsyncStorage** (AsyncStorage is not for binary blobs)
- New "Gym" tab with record flow, attempts list, and per-attempt playback
- Storage management: total-size display, delete individual attempt (removes file + metadata), clear all
- Graceful camera/microphone permission-denial handling

### Data model

Video binaries live on the filesystem; AsyncStorage holds lightweight metadata only.

```ts
// src/store/attemptStore.ts
export type AttemptResult = 'send' | 'attempt';

export interface GymAttempt {
  id: string;             // `attempt-${ts}-${rand}`
  videoUri: string;       // file:// path under FileSystem.documentDirectory + 'attempts/'
  fileSizeBytes: number;
  durationSec: number;
  recordedAt: number;     // epoch ms — source of truth for date and session grouping
  gymName: string;        // free text
  grade: string;          // free text, e.g. "V4"
  result: AttemptResult;
}
```

Zustand + `persist(AsyncStorage)`, mirroring `maxHangStore.ts`. Actions: `addAttempt`, `deleteAttempt` (deletes the file first, then the record), `clearAttempts` (deletes all files). Cap the list (e.g. 100) and evict oldest, deleting their files.

**"Gym session" grouping is derived, not a separate persisted entity** — the attempts list groups by `gymName` + calendar day of `recordedAt`. This satisfies the "associated with a session entry" AC without a second store, keeping the footprint minimal. (Alternative — an explicit `GymSession` object — is deferred unless a follow-up needs cross-session identity.)

### File storage

- Recordings are written by `expo-camera` to the cache, then moved to `FileSystem.documentDirectory + 'attempts/<id>.mov'` so they survive cache eviction.
- `deleteAttempt` / `clearAttempts` remove the file via `FileSystem.deleteAsync(uri, { idempotent: true })` before/after mutating the store so no orphaned files or dangling URIs remain.
- Total storage size = sum of `fileSizeBytes`, shown in the list header and formatted with a `formatBytes` helper.

### Screens (`app/(tabs)/`)

New tab `gym.tsx` (icon 🎥) added to `_layout.tsx`. Flow:

1. **Record view** — `expo-camera` `CameraView` (`mode="video"`), start/stop button, front/back toggle, elapsed-time readout. On stop → metadata form.
2. **Metadata form** — gym name (text), grade (text), result toggle (Send / Attempt), Save / Discard. Save moves the file and calls `addAttempt`; Discard deletes the temp file.
3. **Attempts list** — grouped by gym + day, each row shows date, grade, result badge, size; tap → playback; swipe/long-press → delete (confirm `Alert`). Empty state mirrors `history.tsx`.
4. **Playback** — `expo-video` `VideoView` in a modal/screen (`expo-av`'s `Video` is deprecated in SDK 56; use `expo-video`).

Reuse the existing dark palette (`#111827` / `#1F2937` / `#374151` / `#3B82F6`) and card styling from `history.tsx`.

### Permissions

- `useCameraPermissions()` and `useMicrophonePermissions()` from `expo-camera`. Before recording, request both; if denied, render an explanatory view with a "Open Settings" button (`Linking.openSettings()`) instead of the camera. No crash, no silent failure.
- `app.json`: add the `expo-camera` config plugin with `cameraPermission` / `microphonePermission` (and record-audio) strings. Because native permissions change, this requires a **dev build / prebuild** — it will not run in a bare Expo Go client. Note in the PR.

### i18n

Extend `AppStrings` in `src/i18n/strings.ts` with EN + JA keys for all new labels (record, stop, gym name, grade, result, send, attempt, save, discard, delete, no-attempts empty state, permission-denied copy, storage size).

## Acceptance Criteria

- [ ] User can record a video of a boulder attempt from within the app (start/stop) on iOS and Android via a dev build
- [ ] On stopping, the user enters gym name, grade, and result (send/attempt); the attempt is saved with an auto date/time
- [ ] Saved attempts appear in a list grouped by gym + day, showing date, grade, and result
- [ ] User can play back a saved recording
- [ ] User can delete a saved recording, and both the metadata record **and** the underlying video file are removed
- [ ] WHEN camera or microphone permission is denied, the app SHALL show an explanatory screen with a settings shortcut and SHALL NOT crash (EARS)
- [ ] WHEN available storage cannot hold a recording or a file write fails, the app SHALL surface an error to the user and SHALL NOT leave a dangling metadata record (EARS)
- [ ] Video files are stored on the filesystem; AsyncStorage holds only metadata (verified by inspecting persisted keys)
- [ ] All new user-facing strings are available in EN and JA

## Proposed Approach

**Key components touched / added**

| File | Change |
|------|--------|
| `package.json` / `app.json` | Add `expo-camera`, `expo-file-system`, `expo-video` (SDK 56 aligned); add `expo-camera` plugin + permission strings |
| `src/store/attemptStore.ts` | **New** — Zustand persist store, filesystem-aware add/delete/clear |
| `src/services/videoStorage.ts` | **New** — file move/delete/size helpers wrapping `expo-file-system` |
| `app/(tabs)/gym.tsx` | **New** — record → metadata → list → playback flow |
| `app/(tabs)/_layout.tsx` | Register the new tab |
| `src/i18n/strings.ts` | EN + JA keys for the feature |

**Sequence**

1. Add dependencies + `app.json` plugin/permissions.
2. Build `videoStorage.ts` + `attemptStore.ts` (pure logic first — grouping, `formatBytes`, id/eviction).
3. Build the record view + permission gating.
4. Build metadata form → save/discard wired to store + filesystem.
5. Build grouped list + delete + storage-size header.
6. Build playback view.
7. Add i18n strings; wire the tab.

## Tasks

- [ ] Add `expo-camera`, `expo-file-system`, `expo-video`; configure `app.json` plugin + iOS/Android permission strings
- [ ] `src/services/videoStorage.ts` — move-to-documents, delete, size, `formatBytes`
- [ ] `src/store/attemptStore.ts` — persist store + `addAttempt` / `deleteAttempt` / `clearAttempts` + derived grouping selector
- [ ] Record view with start/stop, front/back toggle, elapsed timer, permission gating
- [ ] Metadata form (gym, grade, result) with Save / Discard
- [ ] Attempts list grouped by gym + day, result badge, per-attempt delete (confirm), total-size header
- [ ] Playback view (`expo-video`)
- [ ] EN + JA i18n strings
- [ ] Register `gym` tab in `_layout.tsx`
- [ ] Add `jest-expo` test setup; unit-test pure logic (grouping, `formatBytes`, store add/delete reducers, eviction) — test-first
- [ ] Manual QA on iOS + Android dev build: record, save, list, play, delete, permission-denial

## Dependencies

- None (foundation feature). Follow-ups (route identification, auto-edit/post) depend on this issue.
- **Toolchain note:** native permissions require a dev/prebuild client — the feature cannot be exercised in stock Expo Go.

## Open Decisions (recommend, not blocking)

1. **Session grouping** — recommend *derived* grouping (gym + day) over a persisted `GymSession` store for this first pass. Revisit if a follow-up needs stable session IDs.
2. **Playback library** — recommend `expo-video` over the already-installed but deprecated `expo-av` `Video`. If avoiding a new dep is preferred, `expo-av` `Video` is an acceptable fallback.

## Research Review

Not applicable — video capture/storage is standard mobile app functionality, outside the `weyucou/ml-playbook` covered domains.
